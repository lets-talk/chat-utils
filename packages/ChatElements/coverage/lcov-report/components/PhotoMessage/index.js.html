<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/PhotoMessage/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/PhotoMessage</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">23.08% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>3/13</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/4</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
&nbsp;
import FaCloudDownload from 'react-icons/lib/fa/cloud-download';
&nbsp;
import './index.scss';
&nbsp;
const PhotoMessage = (props) =&gt; {
  return (
    &lt;div className={classNames('letstalk-mbox-attachment', 'letstalk-mbox-photo')}&gt;
      &lt;div
        className="letstalk-mbox-photo--img"
        style={props.data.width &amp;&amp; <span class="branch-1 cbranch-no" title="branch not covered" >props.data.height </span>&amp;&amp; <span class="branch-2 cbranch-no" title="branch not covered" >{</span>
          width: props.data.width,
          height: props.data.height,
        }}
      &gt;
        &lt;img
          role="button"
          src={props.data.uri}
          alt={props.data.alt}
          onClick={props.onOpen}
          onKeyPress={props.onOpen}
        /&gt;
        {
          props.data.status &amp;&amp;
<span class="branch-1 cbranch-no" title="branch not covered" >            !props.data.status.download </span>&amp;&amp;
<span class="branch-2 cbranch-no" title="branch not covered" >            &lt;div className="letstalk-mbox-photo--img__block"&gt;</span>
              {
                !props.data.status.click &amp;&amp;
                &lt;button
                  onClick={props.onDownload}
                  className="letstalk-mbox-photo--img__block-item letstalk-mbox-photo--download"
                &gt;
                  &lt;FaCloudDownload /&gt;
                &lt;/button&gt;
              }
              {
                typeof props.data.status.loading === 'number' &amp;&amp;
                    props.data.status.loading !== 0 &amp;&amp;
                    &lt;div /&gt;
              }
            &lt;/div&gt;
        }
      &lt;/div&gt;
      {
        props.text &amp;&amp;
<span class="branch-1 cbranch-no" title="branch not covered" >        &lt;div className="letstalk-mbox-text"&gt;</span>
          {props.text}
        &lt;/div&gt;
      }
    &lt;/div&gt;
  );
};
&nbsp;
PhotoMessage.defaultProps = {
  text: '',
  data: {},
  onDownload: null,
  onOpen: null,
};
&nbsp;
PhotoMessage.propTypes = {
  text: PropTypes.string,
  data: PropTypes.object,
  onDownload: PropTypes.func,
  onOpen: PropTypes.func,
};
&nbsp;
export default PhotoMessage;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Nov 24 2017 01:32:25 GMT-0300 (-03)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
