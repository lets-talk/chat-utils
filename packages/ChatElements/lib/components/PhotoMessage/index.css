.letstalk-mbox-photo {
    margin-top: -3px;
    margin-right: -6px;
    margin-left: -6px;
}

.letstalk-mbox-photo .letstalk-mbox-text {
    padding: 5px 0px;
    max-width: 300px;
    margin: auto;
}

.letstalk-mbox-photo--img {
    position: relative;
    display: flex;
    overflow: hidden;
    justify-content: center;
    border-radius: 5px;
    max-height: 300px;
}

.letstalk-mbox-photo--img__block {
    position: absolute;
    top: 0;
    right: 0;
    left: 0;
    bottom: 0;
    background-color: rgba(0,0,0,0.5);
    border-radius: 5px;
    display: flex;
}

.letstalk-mbox-photo--img img {
    height: 100%;
    min-height: 100px;
    user-select: none;
}

.letstalk-mbox-photo--img__block-item {
    margin: auto;
    cursor: pointer;
}

.letstalk-mbox-photo--download {
    color: #efe;
    display: flex;
    justify-content: center;
    background: none;
    border: none;
    font-size: 3.2em;
    outline: none;
    border: 1px solid #eee;
    border-radius: 100%;
    height: 100px;
    width: 100px;
}

.letstalk-mbox-photo--download:hover {
    opacity: .7;
}

.letstalk-mbox-photo--download:active {
    opacity: .3;
}
