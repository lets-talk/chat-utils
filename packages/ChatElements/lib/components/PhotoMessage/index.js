import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import FaCloudDownload from 'react-icons/lib/fa/cloud-download';

import './index.scss';

const PhotoMessage = (props) => (
  <div className={classNames('letstalk-mbox-attachment', 'letstalk-mbox-photo')}>
    <div
      className="letstalk-mbox-photo--img"
      style={props.data.width && props.data.height && {
        width: props.data.width,
        height: props.data.height,
      }}
    >
      <img
        tabIndex={0}
        role="button"
        src={props.data.uri}
        alt={props.data.alt}
        onClick={props.onOpen}
        onKeyPress={props.onOpen}
      />
      {
        props.data.status &&
          !props.data.status.download &&
          <div className="letstalk-mbox-photo--img__block">
            {
              !props.data.status.click &&
              <button
                onClick={props.onDownload}
                className="letstalk-mbox-photo--img__block-item letstalk-mbox-photo--download"
              >
                <FaCloudDownload />
              </button>
            }
            {
              typeof props.data.status.loading === 'number' &&
                  props.data.status.loading !== 0 &&
                  <div />
            }
          </div>
      }
    </div>
    {
      props.text &&
      <div className="letstalk-mbox-text">
        {props.text}
      </div>
    }
  </div>
);

PhotoMessage.defaultProps = {
  text: '',
  data: {},
  onDownload: null,
  onOpen: null,
};

PhotoMessage.propTypes = {
  text: PropTypes.string,
  data: PropTypes.object,
  onDownload: PropTypes.func,
  onOpen: PropTypes.func,
};

export default PhotoMessage;
