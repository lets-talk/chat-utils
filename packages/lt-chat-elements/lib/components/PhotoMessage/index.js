import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import FaCloudDownload from 'react-icons/lib/fa/cloud-download';

import './index.scss';

const PhotoMessage = (props) => (
  <div className={classNames('letstalk-mbox-attachment', 'letstalk-mbox-photo')}>
    <div
      tabIndex="0"
      role="button"
      className="letstalk-mbox-photo--img"
      style={props.data.width && props.data.height && {
        width: props.data.width,
        height: props.data.height,
      }}
      onClick={props.onOpen}
      onKeyPress={props.onOpen}
    >
      <img
        src={props.data.uri}
        alt={props.data.alt}
      />
      {
        props.data.status &&
          !props.data.status.download &&
          <div className="letstalk-mbox-photo--img__block">
            {
              !props.data.status.click &&
              <button
                onClick={props.onDownload}
                className="letstalk-mbox-photo--img__block-item letstalk-mbox-photo--download"
              >
                <FaCloudDownload />
              </button>
            }
            {
              typeof props.data.status.loading === 'number' &&
                  props.data.status.loading !== 0 &&
                  <div>{props.data.status.loading}</div>
            }
          </div>
      }
    </div>
    {
      props.text &&
      <div className="letstalk-mbox-text">
        {props.text}
      </div>
    }
  </div>
);

PhotoMessage.defaultProps = {
  text: '',
  data: {},
  onDownload: null,
  onOpen: null,
};

PhotoMessage.propTypes = {
  text: PropTypes.string,
  data: PropTypes.object,
  onDownload: PropTypes.func,
  onOpen: PropTypes.func,
};

export default PhotoMessage;
