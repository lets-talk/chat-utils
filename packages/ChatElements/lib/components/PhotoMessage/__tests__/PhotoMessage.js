import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PhotoMessage from '../index';

describe('PhotoMessage component', () => {
  it('should render a System message', () => {
    const props = {
      text: 'I am a photo message',
      data: {
        width: 50,
        height: 50,
        uri: '',
        alt: 'Image alt',
      },
    };
    const component = shallow(<PhotoMessage {...props} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('Test clickHandler event with Promise', () => {
    const mockClickHandler = jest.fn(() => Promise.resolve({}));
    const fakeEventObject = { preventDefault() {}, stopPropagation() {} };

    const props = {
      data: {
        width: 50,
        height: 50,
        uri: '',
        alt: 'Image alt',
        status: {
          loading: 70,
        },
      },
    };

    const button = shallow((<PhotoMessage onOpen={mockClickHandler} {...props} />));
    button.find('.letstalk-mbox-photo--img').simulate('click', fakeEventObject);
    expect(mockClickHandler.mock.calls.length).toEqual(1);
  });

  it('Test clickHandler event with Promise', () => {
    const mockClickHandler = jest.fn(() => Promise.resolve({}));
    const fakeEventObject = { preventDefault() {}, stopPropagation() {} };

    const props = {
      data: {
        width: 50,
        height: 50,
        uri: '',
        alt: 'Image alt',
        status: {
          download: true,
        },
      },
    };

    const button = shallow((<PhotoMessage onDownload={mockClickHandler} {...props} />));
    button.find('.letstalk-mbox-photo--download').simulate('click', fakeEventObject);
    expect(mockClickHandler.mock.calls.length).toEqual(1);
  });
});
