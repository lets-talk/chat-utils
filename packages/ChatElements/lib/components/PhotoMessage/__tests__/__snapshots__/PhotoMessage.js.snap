// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoMessage component should render without issues 1`] = `
<div
  className="letstalk-mbox-attachment letstalk-mbox-photo"
>
  <div
    className="letstalk-mbox-photo--img"
  >
    <img
      onClick={null}
      onKeyPress={null}
      role="button"
    />
  </div>
</div>
`;
